fe1-web/src/core/components/__tests__/QrCode.test.tsx

```diff
@@ -0,0 +1,43 @@
+import { render, waitFor } from '@testing-library/react-native';
+import React from 'react';
+
+import QRCode from '../QRCode';
+
+const shortText = '8eFmp65HHy7HjJjfTVqfx9t6x5xNqG6MaqRJ2xQaSPvs5vSj5j5RQjU5aS';
+const longText = `${shortText}x`;
+
+// The type is the one of the warn function
+let tmpConsoleWarn: {
+  (...data: any[]): void;
+  (message?: any, ...optionalParams: any[]): void;
+  (message?: any, ...optionalParams: any[]): void;
+  (...data: any[]): void;
+  (message?: any, ...optionalParams: any[]): void;
+  (message?: any, ...optionalParams: any[]): void;
+};
+describe('QrCode ', () => {
+  // It is useless to test the contrary since we would wait and validate instantly even it would be wrong
+  it('throws a warning, short text with overlay', () => {
+    tmpConsoleWarn = global.console.warn;
+    global.console.warn = jest.fn();
+
+    render(<QRCode value={shortText} overlayText="Overlay text" />);
+    waitFor(() =>
+      expect(global.console.warn).toHaveBeenCalledWith(
+        'An overlay text has been added on a QRCode whose represents a too short text (length < 50)',
+      ),
+    );
+
+    global.console.warn = tmpConsoleWarn;
+  });
+
+  it('looks good with short text and an overlay', () => {
+    const { toJSON } = render(<QRCode value={shortText} overlayText="OverlayText" />);
+    expect(toJSON()).toMatchSnapshot();
+  });
```

Comment:
In my opinion, it seems a bit weird to test the visual of a case we would like to avoid (short text with overlay text)

Reply:
Haha I agree but with my comment above this test would make sense with a slightly different description. Maybe @MeKHell already had something like this in mind :)

Reply:
The idea of this test is :

- Make sure that even if we do not guarantee the QR Code is readable, it still looks good in terms of UI. (For the QR Code readability, we test it through warning test)
- Have a test that checks the blue overlay circle